### ISSUE TITLE
doc: Missing some snapshot format for demos

### ISSUE THREAD

**Issue by heinrich5991** [heinrich5991/libtw2#81]
Apparently there's some undocumented storage format for snapshots that is used in demos.

**Comment by heinrich5991** [comment]
No, apparently there's some snapshot format that is not snapshot diffs, in the demos. I haven't investigated.

**Comment by Patiga** [comment]
In demos, there are not just snapshot deltas, but also 'pure' snapshots.
I implemented a `SnapReader` for the demo rewrite pull request
https://github.com/heinrich5991/libtw2/pull/74/files#diff-b4a58dccefe5b7c41609f39d5c041da12b89d64ab58344fe31a30ba6c2f75fb6

So first a header with data_size and num_items.
Then, for `num_items` one i32 offset, where the first offset is 0 and monotonically increasing.
The offsets are in bytes, and always multiples of 4.
They are also always > 4, since the first i32 contains the type id and id of the item.
The rest of the data is just the item data.

Hm, intended to expand the demo docs once I finish the rewrite, but it kinda stalled since I paused coding my renderer.
If no one does it, I'll get back to it eventually, hopefully soon.
